browser/ui/webui/settings/brave_sync_handler.cc

```diff
@@ -0,0 +1,217 @@
+  return result;
+}
+
+}  // namespace
+
+BraveSyncHandler::BraveSyncHandler() { }
+BraveSyncHandler::~BraveSyncHandler() { }
+
+void BraveSyncHandler::RegisterMessages() {
+  profile_ = Profile::FromWebUI(web_ui());
+  web_ui()->RegisterMessageCallback(
+      "SyncGetDeviceList",
+       base::BindRepeating(&BraveSyncHandler::HandleGetDeviceList,
+                           base::Unretained(this)));
+  web_ui()->RegisterMessageCallback(
+      "SyncSetupSetSyncCode",
+      base::BindRepeating(&BraveSyncHandler::HandleSetSyncCode,
+                          base::Unretained(this)));
+  web_ui()->RegisterMessageCallback(
+      "SyncSetupGetSyncCode",
+      base::BindRepeating(&BraveSyncHandler::HandleGetSyncCode,
+                          base::Unretained(this)));
+  web_ui()->RegisterMessageCallback(
+      "SyncGetQRCode",
+      base::BindRepeating(&BraveSyncHandler::HandleGetQRCode,
+                          base::Unretained(this)));
+  web_ui()->RegisterMessageCallback(
+      "SyncSetupReset", base::BindRepeating(&BraveSyncHandler::HandleReset,
+                                            base::Unretained(this)));
+}
+
+void BraveSyncHandler::OnJavascriptAllowed() {
+  syncer::DeviceInfoTracker* tracker =
+      DeviceInfoSyncServiceFactory::GetForProfile(profile_)
+          ->GetDeviceInfoTracker();
+  DCHECK(tracker);
+  if (tracker)
+    device_info_tracker_observer_.Add(tracker);
+}
+
+void BraveSyncHandler::OnJavascriptDisallowed() {
+  device_info_tracker_observer_.RemoveAll();
+}
+
+void BraveSyncHandler::OnDeviceInfoChange() {
+  if (IsJavascriptAllowed())
+    FireWebUIListener("device-info-changed", GetSyncDeviceList());
+}
+
+void BraveSyncHandler::HandleGetDeviceList(const base::ListValue* args) {
+  AllowJavascript();
+  const auto& list = args->GetList();
+  CHECK_EQ(1U, list.size());
+  const base::Value* callback_id;
+  CHECK(args->Get(0, &callback_id));
+
+  ResolveJavascriptCallback(*callback_id, GetSyncDeviceList());
+}
+
+void BraveSyncHandler::HandleGetSyncCode(const base::ListValue* args) {
+  AllowJavascript();
+
+  CHECK_EQ(1U, args->GetSize());
+  const base::Value* callback_id;
+  CHECK(args->Get(0, &callback_id));
+
+  brave_sync::Prefs brave_sync_prefs(profile_->GetPrefs());
+  std::string sync_code = brave_sync_prefs.GetSeed();
+  if (sync_code.empty()) {
+    std::vector<uint8_t> seed = brave_sync::crypto::GetSeed();
+    sync_code = brave_sync::crypto::PassphraseFromBytes32(seed);
+  }
+
+  ResolveJavascriptCallback(*callback_id, base::Value(sync_code));
+}
+
+void BraveSyncHandler::HandleGetQRCode(const base::ListValue* args) {
+  AllowJavascript();
+  CHECK_EQ(2U, args->GetSize());
+  const base::Value* callback_id;
+  CHECK(args->Get(0, &callback_id));
+  const base::Value* sync_code;
+  CHECK(args->Get(1, &sync_code));
+
+  std::vector<uint8_t> seed;
+  if (!brave_sync::crypto::PassphraseToBytes32(sync_code->GetString(), &seed)) {
+    LOG(ERROR) << "invalid sync code:" << sync_code->GetString();
```

Comment:
this is only logging invalid codes, but a user might just be using the wrong sync chain or something so we probably should log this

Reply:
yea i think this could be sensitive info and shouldn't be logged

Reply:
fixed in https://github.com/brave/brave-core/pull/5294/commits/247a37c0aca0207366827791456801efcd92cebe